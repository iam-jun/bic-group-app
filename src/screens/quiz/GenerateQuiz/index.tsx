export { default as SubmitFormQuiz } from './SubmitFormQuiz';
export { default as ComposeQuiz } from './ComposeQuiz';
