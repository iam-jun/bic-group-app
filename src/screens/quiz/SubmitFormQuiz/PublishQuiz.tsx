/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable unused-imports/no-unused-imports */
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import React, { FC } from 'react';
import { StyleSheet, View } from 'react-native';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import Header from '~/beinComponents/Header';
import { spacing } from '~/theme';
import useGenerateQuiz from './hooks/useGenerateQuiz';
import TitleDescriptionSection from './components/TitleDescriptionSection';
import QuestionAnswerSection from './components/QuestionAnswerSection';
import QuestionAnswerDisplaySection from './components/QuestionAnswerDisplaySection';
import CheckmarkGenerateRandomQuiz from './components/CheckmarkGenerateRandomQuiz';
import { useBackPressListener } from '~/hooks/navigation';
import useQuizzesStore from '~/store/entities/quizzes';
import { FormGenerateQuiz } from '~/interfaces/IQuiz';

type PublishQuizProps = {
  route?: {
    params?: {
      quizId: string;
    };
  };
};

const PublishQuiz: FC<PublishQuizProps> = (props) => {
  const { route } = props;
  const { params } = route || {};
  const { quizId } = params || {};

  const quiz = useQuizzesStore((state) => state.data[quizId]);
  const { contentId, numberOfQuestions } = quiz || {};

  const initFormData: FormGenerateQuiz = {
    numberOfQuestions,
  };
  const {
    control,
    onPublish,
    isFormValid,
    loading,
    watch,
    trigger,
    handleBack,
  } = useGenerateQuiz(contentId, initFormData);

  const disabled = !isFormValid || loading;

  const theme = useTheme();
  const styles = createStyle(theme);

  useBackPressListener(handleBack);

  return (
    <View style={styles.container}>
      <Header
        useI18n
        title="quiz:quiz_information"
        buttonProps={{
          disabled,
          loading,
          style: styles.btnCreate,
        }}
        buttonText="common:btn_create"
        onPressButton={onPublish(quizId)}
        onPressBack={handleBack}
      />
      <KeyboardAwareScrollView
        style={styles.container}
        contentContainerStyle={styles.containerContent}
      >
        <TitleDescriptionSection control={control} />
        <QuestionAnswerDisplaySection
          control={control}
          watch={watch}
          trigger={trigger}
        />
        <CheckmarkGenerateRandomQuiz />
      </KeyboardAwareScrollView>
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;
  const insets = useSafeAreaInsets();

  return StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: colors.white,
    },
    containerContent: {
      paddingHorizontal: spacing.padding.large,
      paddingTop: spacing.padding.large,
      paddingBottom: spacing.padding.large + insets.bottom,
    },
    btnCreate: {
      marginRight: spacing.margin.small,
    },
  });
};

export default PublishQuiz;
