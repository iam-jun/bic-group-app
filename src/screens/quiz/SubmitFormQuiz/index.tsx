export { default as EntryQuiz } from './EntryQuiz';
export { default as PublishQuiz } from './PublishQuiz';
