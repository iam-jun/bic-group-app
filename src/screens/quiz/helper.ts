import i18next from 'i18next';
import usePostsStore from '~/store/entities/posts';
import useQuizzesStore from '~/store/entities/quizzes';

export const mapIndexToAlphabet = ['A', 'B', 'C', 'D', 'E', 'F'];

export const handleQuizNotificationSocket = (msg: any) => {
  const { activities = [] } = msg || {};
  const { quizInfo } = activities[0] || {};
  const { quizId, contentId } = quizInfo || {};

  const { waitingProcessingQuiz, actions: actionsQuizzesStore } = useQuizzesStore.getState();

  // if user is still standing in ComposeQuiz screen to wait for generating quiz
  // then getQuizDetail & getPostDetail for getting the new quiz's status
  if (waitingProcessingQuiz === quizId) {
    actionsQuizzesStore.getQuizDetail({ quizId, isShowLoading: false });
    usePostsStore.getState().actions.getPostDetail({ postId: contentId });
  }
};

export const validateSpaceTrap = (value) => value.trim().length > 0;

export const validateIntegerNumber = (value): any => Number.isInteger(Number(value)) || i18next.t('quiz:enter_an_integer');
