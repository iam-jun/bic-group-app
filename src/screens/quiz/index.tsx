export { EntryQuiz, PublishQuiz } from './SubmitFormQuiz';
export { default as ComposeQuiz } from './ComposeQuiz';
export { default as EditQuestion } from './EditQuestion';
export { default as TakeQuiz } from './TakeQuiz';
export { default as TakeQuizReview } from './TakeQuizReview';
export { default as TakeQuizResult } from './TakeQuizResult';
export { default as PreviewDraftQuizNotification } from './PreviewDraftQuizNotification';
export { default as YourQuiz } from './YourQuiz';
