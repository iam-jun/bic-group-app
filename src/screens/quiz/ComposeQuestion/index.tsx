import React, { FC, useRef } from 'react';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import { StyleSheet, View } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { FormProvider, useFieldArray } from 'react-hook-form';
import { isNumber } from 'lodash';
import useQuizzesStore from '~/store/entities/quizzes';
import Header from '~/beinComponents/Header';
import { useBaseHook } from '~/hooks';
import useComposeQuestion, {
  ValidateComposeQuestionError,
} from './hooks/useComposeQuestion';
import { spacing } from '~/theme';
import QuestionField from './components/QuestionField';
import AnswerField from './components/AnswerField';
import { useBackPressListener } from '~/hooks/navigation';
import { MAX_ANSWERS } from '../helper';
import { Button } from '~/baseComponents';
import Icon from '~/baseComponents/Icon';
import Text from '~/baseComponents/Text';

type ComposeQuestionProps = {
  route?: {
    params?: {
      questionIndex?: number;
      quizId: string;
    };
  };
};

const ComposeQuestion: FC<ComposeQuestionProps> = (props) => {
  const { route } = props;
  const { params } = route || {};
  const { questionIndex, quizId } = params || {};
  const isCreatingQuestion = !isNumber(questionIndex);

  const scrollViewRef = useRef<any>();

  const loading = useQuizzesStore((state) => state.loading);
  const quiz = useQuizzesStore((state) => state.data[quizId]);
  const { questions = [] } = quiz || {};

  const {
    methods,
    enabledBtnSave,
    needToChooseCorrectAnswer,
    onSave,
    onRemoveQuestion,
    handleBack,
  } = useComposeQuestion(quiz, questionIndex);

  const { control } = methods || {};

  const { fields, remove, append } = useFieldArray({
    control,
    name: 'answers',
  });

  const { t } = useBaseHook();
  const theme = useTheme();
  const { colors } = theme;
  const styles = createStyle(theme);

  const inputsYPosition = useRef({});

  const onLayoutAnswerView = (indexInputAnswer: number, y: number) => {
    inputsYPosition.current[indexInputAnswer] = y;
  };

  const onPressAddAnswer = () => {
    append({
      content: '',
      isCorrect: false,
    });
  };

  const onValidateError: ValidateComposeQuestionError = (
    type: 'question' | 'answers',
    indexInputAnswer?: number,
  ) => {
    switch (type) {
      case 'question':
        // scroll to question field
        scrollViewRef?.current?.scrollToPosition?.(0, 0, true);
        break;
      case 'answers':
        if (isNumber(indexInputAnswer)) {
          // scroll to answer field
          scrollViewRef?.current?.scrollToPosition?.(
            0,
            inputsYPosition.current[indexInputAnswer],
            true,
          );
        } else {
          // scroll to end
          scrollViewRef?.current?.scrollToEnd?.(true);
        }
        break;
    }
  };

  useBackPressListener(handleBack);

  return (
    <View style={styles.container}>
      <Header
        useI18n
        title={
          isCreatingQuestion
            ? t('quiz:add_question')
            : t('quiz:edit_the_question_number', {
              number: questionIndex + 1,
            })
        }
        buttonProps={{
          disabled: !enabledBtnSave,
          loading,
          style: styles.btnSave,
        }}
        buttonText="common:btn_save"
        onPressButton={onSave(onValidateError)}
        onPressBack={handleBack}
      />
      <KeyboardAwareScrollView
        ref={scrollViewRef}
        style={styles.scrollView}
        contentContainerStyle={styles.contentContainer}
        alwaysBounceVertical={false}
        keyboardShouldPersistTaps="handled"
      >
        <FormProvider {...methods as any}>
          <QuestionField
            remove={onRemoveQuestion}
            questionIndex={
              isCreatingQuestion ? questions.length : questionIndex
            }
          />
          {fields.map((field, index) => (
            <AnswerField
              key={`answer_field.${field.id}`}
              remove={remove}
              answerIndex={index}
              onLayoutAnswerView={onLayoutAnswerView}
            />
          ))}
          {fields.length < MAX_ANSWERS && (
            <Button.Neutral
              type="ghost"
              icon="Plus"
              iconSize={16}
              useI18n
              style={styles.btnAddAnswer}
              onPress={onPressAddAnswer}
            >
              quiz:add_answer
            </Button.Neutral>
          )}
          {needToChooseCorrectAnswer && (
            <View style={styles.errorView}>
              <Icon
                icon="CircleExclamation"
                size={14}
                tintColor={colors.red40}
              />
              <Text.BodyXS style={styles.errorText} useI18n>
                quiz:please_select_correct_answer
              </Text.BodyXS>
            </View>
          )}
        </FormProvider>
      </KeyboardAwareScrollView>
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: colors.white,
    },
    btnSave: {
      marginRight: spacing.margin.small,
    },
    scrollView: {
      flex: 1,
    },
    contentContainer: {
      padding: spacing.padding.large,
      paddingBottom: spacing.padding.large + 84,
    },
    btnAddAnswer: {
      alignSelf: 'baseline',
      marginTop: spacing.margin.small,
    },
    errorView: {
      flexDirection: 'row',
      flex: 1,
      marginTop: spacing.margin.small,
    },
    errorText: {
      color: colors.red40,
      marginLeft: spacing.margin.tiny,
    },
  });
};

export default ComposeQuestion;
