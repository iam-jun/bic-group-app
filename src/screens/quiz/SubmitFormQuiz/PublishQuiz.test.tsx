import React from 'react';
import useQuizzesStore from '~/store/entities/quizzes';
import MockedNavigator from '~/test/MockedNavigator';
import { mockGenerateQuizResponse } from '~/test/mock_data/quiz';
import {
  fireEvent, renderHook, renderWithRedux, waitFor,
} from '~/test/testUtils';
import PublishQuiz from './PublishQuiz';
import { IQuiz, QuizStatus } from '~/interfaces/IQuiz';
import streamApi from '~/api/StreamApi';

describe('PublishQuiz', () => {
  it('should render correctly', async () => {
    const { result } = renderHook(() => useQuizzesStore());

    result.current.actions.addOrUpdateQuiz(
      mockGenerateQuizResponse.data as IQuiz,
    );

    const wrapper = renderWithRedux(
      <MockedNavigator
        component={() => (
          <PublishQuiz
            route={{
              params: { quizId: mockGenerateQuizResponse.data.id },
            }}
          />
        )}
      />,
    );

    await waitFor(() => {
      wrapper.getByTestId('title_description_section.title');
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should publish successfully', async () => {
    const fakePublishQuizSuccess = {
      ...mockGenerateQuizResponse,
      data: {
        ...mockGenerateQuizResponse.data,
        status: QuizStatus.PUBLISHED,
        title: 'test',
      },
    };
    const spyApiEditQuiz = jest
      .spyOn(streamApi, 'editQuiz')
      .mockImplementation(() => Promise.resolve(fakePublishQuizSuccess) as any);

    const { result } = renderHook(() => useQuizzesStore());

    result.current.actions.addOrUpdateQuiz(
      mockGenerateQuizResponse.data as IQuiz,
    );

    const wrapper = renderWithRedux(
      <MockedNavigator
        component={() => (
          <PublishQuiz
            route={{
              params: { quizId: mockGenerateQuizResponse.data.id },
            }}
          />
        )}
      />,
    );

    let inputTitle;
    await waitFor(() => {
      inputTitle = wrapper.getByTestId('title_description_section.title');
    });

    fireEvent.changeText(inputTitle, 'test');

    const btnCreate = wrapper.getByTestId('header.button');
    await waitFor(() => {
      expect(btnCreate.props.disabled).toBeFalsy();
    });

    fireEvent.press(btnCreate);

    await waitFor(() => {
      const params = spyApiEditQuiz.mock.calls[0][1] || {};
      const { title, description, numberOfQuestionsDisplay } = params;

      expect(title).toBe('test');
      expect(description).toBe(null);
      expect(numberOfQuestionsDisplay).toBeUndefined();
    });

    await waitFor(() => {
      expect(result.current.data[mockGenerateQuizResponse.data.id].title).toBe('test');
      expect(result.current.data[mockGenerateQuizResponse.data.id].status).toBe(QuizStatus.PUBLISHED);
    });
  });

  it('should edit successfully', async () => {
    const fakeQuiz = {
      ...mockGenerateQuizResponse,
      data: {
        ...mockGenerateQuizResponse.data,
        status: QuizStatus.PUBLISHED,
        title: 'title1',
        numberOfQuestionsDisplay: 10,
      },
    };
    const fakeEditQuizSuccess = {
      ...mockGenerateQuizResponse,
      data: {
        ...mockGenerateQuizResponse.data,
        status: QuizStatus.PUBLISHED,
        title: 'test',
        numberOfQuestionsDisplay: 10,
      },
    };
    const spyApiEditQuiz = jest
      .spyOn(streamApi, 'editQuiz')
      .mockImplementation(() => Promise.resolve(fakeEditQuizSuccess) as any);

    const { result } = renderHook(() => useQuizzesStore());

    result.current.actions.addOrUpdateQuiz(
      fakeQuiz.data as IQuiz,
    );

    const wrapper = renderWithRedux(
      <MockedNavigator
        component={() => (
          <PublishQuiz
            route={{
              params: { quizId: mockGenerateQuizResponse.data.id },
            }}
          />
        )}
      />,
    );

    let inputTitle;
    await waitFor(() => {
      inputTitle = wrapper.getByTestId('title_description_section.title');
    });

    fireEvent.changeText(inputTitle, 'test');

    const btnSave = wrapper.getByTestId('header.button');
    await waitFor(() => {
      expect(btnSave.props.disabled).toBeFalsy();
    });

    fireEvent.press(btnSave);

    await waitFor(() => {
      const params = spyApiEditQuiz.mock.calls[0][1] || {};
      const { title, description, numberOfQuestionsDisplay } = params;

      expect(title).toBe('test');
      expect(description).toBe(null);
      expect(numberOfQuestionsDisplay).toBe(10);
    });

    await waitFor(() => {
      expect(result.current.data[mockGenerateQuizResponse.data.id].title).toBe('test');
    });
  });
});
