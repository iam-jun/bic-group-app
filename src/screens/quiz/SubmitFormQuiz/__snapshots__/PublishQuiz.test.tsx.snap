// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishQuiz should render correctly 1`] = `
<RNSScreenStack
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNSScreen
    collapsable={false}
    forwardedRef={[Function]}
    gestureResponseDistance={
      Object {
        "bottom": -1,
        "end": -1,
        "start": -1,
        "top": -1,
      }
    }
    nativeBackButtonDismissalEnabled={false}
    onAppear={[Function]}
    onDisappear={[Function]}
    onDismissed={[Function]}
    onHeaderBackButtonClicked={[Function]}
    onTransitionProgress={[Function]}
    onWillAppear={[Function]}
    onWillDisappear={[Function]}
    replaceAnimation="pop"
    stackPresentation="push"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    swipeDirection="horizontal"
  >
    <RNSScreenStackHeaderConfig
      backgroundColor="rgb(255, 255, 255)"
      color="rgb(0, 122, 255)"
      hidden={false}
      onAttached={[Function]}
      onDetached={[Function]}
      title="MockedScreen"
      titleColor="rgb(28, 28, 30)"
      topInsetEnabled={true}
      translucent={false}
    />
    <View
      stackPresentation="push"
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "zIndex": 2,
            }
          }
          testID="header"
        >
          <View
            animatedStyle={
              Object {
                "value": Object {},
              }
            }
            collapsable={false}
            style={
              Object {
                "alignItems": "flex-end",
                "backgroundColor": "#FFFFFF",
                "elevation": 2,
                "flexDirection": "row",
                "minHeight": 44,
                "paddingTop": 0,
                "paddingVertical": 4,
                "shadowColor": "rgba(0, 0, 0, 1)",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.12,
                "shadowRadius": 1.5,
              }
            }
            testID="header.content"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flex": 1,
                  "flexDirection": "row",
                  "overflow": "hidden",
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <View
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                hitSlop={
                  Object {
                    "bottom": 20,
                    "left": 20,
                    "right": 20,
                    "top": 20,
                  }
                }
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="header.back.button"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      },
                      Object {
                        "alignItems": "center",
                        "height": 44,
                        "justifyContent": "center",
                        "padding": 12,
                        "width": 44,
                      },
                      Object {
                        "backgroundColor": undefined,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <View
                      testID="header.back"
                    />
                  </View>
                </View>
              </View>
              <View
                animatedStyle={
                  Object {
                    "value": Object {},
                  }
                }
                collapsable={false}
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                    "paddingVertical": 8,
                  }
                }
              >
                <View
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "color": "#181C32",
                          "fontFamily": "BeVietnamPro-SemiBold",
                          "fontSize": 15,
                        },
                        Object {},
                        Object {
                          "marginRight": 8,
                        },
                      ]
                    }
                    testID="header.text"
                  >
                    Quiz information
                  </Text>
                </View>
              </View>
              <View
                accessibilityState={
                  Object {
                    "disabled": true,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "borderRadius": 8,
                    },
                    Object {
                      "marginRight": 8,
                    },
                  ]
                }
                testID="header.button"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderRadius": 8,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "paddingVertical": 4,
                      },
                      undefined,
                      Object {
                        "backgroundColor": "#F4EFFB",
                        "height": 36,
                        "paddingHorizontal": 12,
                      },
                    ]
                  }
                  testID="button.content"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#212745",
                          "fontFamily": "BeVietnamPro-Medium",
                          "fontSize": 15,
                        },
                        Object {
                          "color": "#BAC0DE",
                        },
                        Array [
                          Object {
                            "textAlign": "center",
                          },
                          undefined,
                        ],
                      ]
                    }
                    testID="button.text"
                  >
                    Create
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "height": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <RCTScrollView
          automaticallyAdjustContentInsets={false}
          contentContainerStyle={
            Object {
              "paddingBottom": 16,
              "paddingHorizontal": 16,
              "paddingTop": 16,
            }
          }
          contentInset={
            Object {
              "bottom": 0,
            }
          }
          enableAutomaticScroll={true}
          enableOnAndroid={false}
          enableResetScrollToCoords={true}
          extraHeight={75}
          extraScrollHeight={0}
          getScrollResponder={[Function]}
          handleOnScroll={[Function]}
          keyboardDismissMode="interactive"
          keyboardOpeningTime={250}
          keyboardSpace={0}
          onScroll={[Function]}
          resetKeyboardSpace={[Function]}
          scrollEventThrottle={1}
          scrollForExtraHeightOnAndroid={[Function]}
          scrollIntoView={[Function]}
          scrollToEnd={[Function]}
          scrollToFocusedInput={[Function]}
          scrollToPosition={[Function]}
          showsVerticalScrollIndicator={true}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "flex": 1,
            }
          }
          update={[Function]}
          viewIsInsideTabBar={false}
        >
          <View>
            <View>
              <View
                style={
                  Array [
                    Object {
                      "flexDirection": "column",
                      "marginVertical": 4,
                    },
                    undefined,
                  ]
                }
                testID="text_input"
              >
                <View
                  style={
                    Object {
                      "marginBottom": 8,
                      "marginRight": 0,
                      "maxWidth": 750,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#353E6E",
                          "fontFamily": "BeVietnamPro-Light",
                          "fontSize": 15,
                        },
                        Object {
                          "color": "#181C32",
                        },
                        undefined,
                      ]
                    }
                  >
                    Title
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flexDirection": "row",
                          "height": 40,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "borderBottomLeftRadius": 6,
                            "borderBottomRightRadius": 6,
                            "borderTopLeftRadius": 6,
                            "borderTopRightRadius": 6,
                            "borderWidth": 1,
                            "flex": 1,
                            "flexDirection": "row",
                          },
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "borderColor": "#E3E6F2",
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "height": 40,
                          }
                        }
                      >
                        <TextInput
                          editable={true}
                          maxLength={64}
                          onBlur={[Function]}
                          onChangeText={[Function]}
                          onFocus={[Function]}
                          placeholder="Enter title of quiz"
                          placeholderTextColor="#BAC0DE"
                          selectionColor="#767D93"
                          style={
                            Array [
                              Object {
                                "flex": 1,
                                "fontFamily": "BeVietnamPro-Light",
                                "fontSize": 14,
                                "height": 40,
                                "paddingHorizontal": 16,
                              },
                              Object {
                                "color": "#181C32",
                              },
                            ]
                          }
                          testID="title_description_section.title"
                          value=""
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "padding": 16,
                    },
                    Object {
                      "paddingHorizontal": 0,
                    },
                  ]
                }
                testID="text_input.text_area"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#353E6E",
                        "fontFamily": "BeVietnamPro-Light",
                        "fontSize": 15,
                      },
                      Object {
                        "color": "#181C32",
                      },
                      Object {
                        "marginBottom": 8,
                      },
                    ]
                  }
                >
                  Description
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#E3E6F2",
                        "borderRadius": 4,
                        "borderWidth": 1,
                        "marginBottom": 6,
                        "minHeight": 84,
                        "paddingHorizontal": 12,
                        "paddingVertical": 6,
                      },
                      Object {
                        "borderColor": "#E3E6F2",
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    maxLength={255}
                    multiline={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="Describe quiz"
                    placeholderTextColor="#BAC0DE"
                    selectionColor="#767D93"
                    style={
                      Array [
                        Object {
                          "color": "#353E6E",
                          "flex": 1,
                          "fontFamily": "BeVietnamPro-Light",
                          "fontSize": 14,
                        },
                        undefined,
                      ]
                    }
                    testID="title_description_section.description"
                    textAlignVertical="top"
                    value=""
                  />
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#212745",
                        "fontFamily": "BeVietnamPro-Light",
                        "fontSize": 11,
                      },
                      Object {
                        "color": "#BAC0DE",
                      },
                      Object {
                        "alignSelf": "flex-end",
                      },
                    ]
                  }
                >
                  0/255
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderTopColor": "#E3E6F2",
                  "borderTopWidth": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "height": 16,
                    "paddingHorizontal": undefined,
                    "paddingVertical": undefined,
                    "width": undefined,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flexDirection": "column",
                      "marginVertical": 4,
                    },
                    undefined,
                  ]
                }
                testID="text_input"
              >
                <View
                  style={
                    Object {
                      "marginBottom": 8,
                      "marginRight": 0,
                      "maxWidth": 750,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#353E6E",
                          "fontFamily": "BeVietnamPro-Light",
                          "fontSize": 15,
                        },
                        Object {
                          "color": "#181C32",
                        },
                        undefined,
                      ]
                    }
                  >
                    Question
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#353E6E",
                            "fontFamily": "BeVietnamPro-Light",
                            "fontSize": 15,
                          },
                          Object {
                            "color": "#BAC0DE",
                          },
                          undefined,
                        ]
                      }
                    >
                       (optional)
                    </Text>
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flexDirection": "row",
                          "height": 40,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "borderBottomLeftRadius": 6,
                            "borderBottomRightRadius": 6,
                            "borderTopLeftRadius": 6,
                            "borderTopRightRadius": 6,
                            "borderWidth": 1,
                            "flex": 1,
                            "flexDirection": "row",
                          },
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "borderColor": "#E3E6F2",
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "height": 40,
                          }
                        }
                      >
                        <TextInput
                          editable={true}
                          keyboardType="number-pad"
                          onBlur={[Function]}
                          onChangeText={[Function]}
                          onFocus={[Function]}
                          placeholder="Enter number 0-{{max}}"
                          placeholderTextColor="#BAC0DE"
                          selectionColor="#767D93"
                          style={
                            Array [
                              Object {
                                "flex": 1,
                                "fontFamily": "BeVietnamPro-Light",
                                "fontSize": 14,
                                "height": 40,
                                "paddingHorizontal": 16,
                              },
                              Object {
                                "color": "#181C32",
                              },
                            ]
                          }
                          testID="question_answer_display_section.question"
                          value=""
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#212745",
                      "fontFamily": "BeVietnamPro-Light",
                      "fontSize": 13,
                    },
                    Object {},
                    Object {
                      "marginTop": 6,
                    },
                  ]
                }
              >
                Enter the number of question will display for member.
              </Text>
              <View
                style={
                  Object {
                    "height": 24,
                    "paddingHorizontal": undefined,
                    "paddingVertical": undefined,
                    "width": undefined,
                  }
                }
              />
            </View>
            <View
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="button_wrapper"
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  hitSlop={
                    Object {
                      "bottom": 10,
                      "left": 10,
                      "right": 10,
                      "top": 10,
                    }
                  }
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                        },
                        undefined,
                        Object {
                          "backgroundColor": undefined,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          undefined,
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      <View />
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "height": undefined,
                      "paddingHorizontal": undefined,
                      "paddingVertical": undefined,
                      "width": 12,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#212745",
                          "fontFamily": "BeVietnamPro-Light",
                          "fontSize": 13,
                        },
                        Object {},
                        undefined,
                      ]
                    }
                  >
                    Do you want to randomize the questions and answers or not?
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </RNSScreen>
</RNSScreenStack>
`;
